src/main/java/org/wildfly/security/auth/server/SecurityDomain.java

```diff
@@ -383,6 +386,11 @@ SecurityIdentity transform(final SecurityIdentity securityIdentity) {
         return Assert.assertNotNull(securityIdentityTransformer.apply(securityIdentity));
     }
 
+    boolean trustsDomain(final SecurityDomain domain) {
+        Assert.checkNotNullParam("domain", domain);
+        return this == domain ? true : trustedSecurityDomain.test(domain);
```

Comment:
This could be simplified: return this == domain || trustedSecurityDomain.test(domain);